cmd/network/verification.go

```diff
@@ -4,164 +4,352 @@ import (
-	vpcSubnetID     string
-	cloudImageID    string
-	instanceType    string
+	// clusterId is the cluster identifier that will be used to query backplane for AWS credentials to build an AWS client
+	clusterId       string
+	region          string
+	subnetId        string
 	securityGroupId string
-	cloudTags       map[string]string
 	debug           bool
-	region          string
-	timeout         time.Duration
-	kmsKeyID        string
-	httpProxy       string
-	httpsProxy      string
-	CaCert          string
+	caCert          string
 	noTls           bool
-	awsProfile      string
 }
 
-func getDefaultRegion() string {
-	val, present := os.LookupEnv(regionEnvVarStr)
-	if present {
-		return val
-	}
-	return regionDefault
-
-}
 func NewCmdValidateEgress() *cobra.Command {
-	config := egressConfig{}
+	e := &egressVerification{}
 
 	validateEgressCmd := &cobra.Command{
 		Use:   "verify-egress",
-		Short: "Verify essential openshift domains are reachable from given subnet ID.",
-		Long:  `Verify essential openshift domains are reachable from given subnet ID.`,
-		Example: `For AWS, ensure your credential environment vars 
-AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY (also AWS_SESSION_TOKEN for STS credentials) 
-are set correctly before execution.
-
-# Verify that essential openshift domains are reachable from a given SUBNET_ID
-osdctl network verify-egress --subnet-id $(SUBNET_ID) --security-group $(SECURITY_GROUP) --region $(AWS_REGION)`,
+		Short: "Verify required URLs required for a fully supportable AWS OSD/ROSA cluster.",
+		Long: `Verify required URLs required for a fully supportable AWS OSD/ROSA cluster.
+
+  This command is an opinionated wrapper around running https://github.com/openshift/osd-network-verifier for SREs.
+  Given an OCM cluster name or id, this command will attempt to automatically detect the security group, subnet, and
+  cluster-wide proxy configuration to run osd-network-verifier's egress verification. The purpose of this check is to
+  verify whether a ROSA cluster's VPC allows for all required external URLs are reachable. The exact cause can vary and
+  typically requires a customer to remediate the issue themselves.
+
+  Docs: https://docs.openshift.com/rosa/rosa_install_access_delete_clusters/rosa_getting_started_iam/rosa-aws-prereqs.html#osd-aws-privatelink-firewall-prerequisites_prerequisites`,
+		Example: `
+  # Run against a cluster registered in OCM
+  ocm-backplane tunnel -D
+  osdctl network verify-egress --cluster-id my-rosa-cluster
+
+  # Run against a cluster registered in OCM with a cluster-wide-proxy
+  ocm-backplane tunnel -D
+  touch cacert.txt
+  osdctl network verify-egress --cluster-id my-rosa-cluster --cacert cacert.txt
+
+  # Override automatic selection of a subnet or security group id
+  ocm-backplane tunnel -D
+  osdctl network verify-egress --cluster-id my-rosa-cluster --subnet-id subnet-abcd --security-group sg-abcd
+
+  # (Not recommended) Run against a specific VPC, without specifying cluster-id
```

Comment:
**thought:** The unrecommended way to execute the command is still at the bottom for the README, maybe we should update it there as well? Although it still works that way.

Reply:
> maybe we should update it there as well?

Just want to clarify what you mean here. I wrote that "unrecommended way" example mostly just to show/explain that it could still be done that way without backplane/OCM. I'm ok with removing it too if you think it's confusing to show, but wanted to make sure I understood your suggestion fully.

Thanks for all the feedback in general!

Reply:
Sorry for the confusing comment. The documentation in `verification.go` is perfectly fine, I meant that the project `README.md` still contains the "not recommended" way: https://github.com/openshift/osdctl/blob/master/README.md#osd-network-verifier

Just a thought though, not sure if it should be changed.

Reply:
Oh I didn't even notice that! Yes, definitely agreed that should be changed.